test/test_docannotate.py

```diff
@@ -425,3 +426,30 @@ def __init__(self, arg: str):
 
     assert Demo.__init__.metadata.annotated_params['arg'].type_name == 'str'
     assert DemoAnn.__init__.metadata.annotated_params['arg'].type_class == str
+
+
+def test_docstring_validators():
```

Comment:
There should be another test case here to verify that the validators work to make sure that parsing them produces the equivalent things as specifying them in a `param` decorator.  It could be just another test case calling a function that has been `@docannotate` with validators and passing it arguments that should/should not pass validation, making sure it works end-to-end.

Reply:
added - 2866f1393